ui/spinner.js

```diff
@@ -203,20 +208,31 @@ return $.widget( "ui.spinner", {
 
 	_draw: function() {
 		var uiSpinner = this.uiSpinner = this.element
-			.addClass( "ui-spinner-input" )
 			.attr( "autocomplete", "off" )
 			.wrap( this._uiSpinnerHtml() )
 			.parent()
 				// add buttons
 				.append( this._buttonHtml() );
 
+		this._addClass( this.uiSpinner, "ui-spinner", "ui-widget ui-widget-content" );
+		this._addClass( "ui-spinner-input" );
+
 		this.element.attr( "role", "spinbutton" );
 
 		// button bindings
-		this.buttons = uiSpinner.find( ".ui-spinner-button" )
+		this.buttons = uiSpinner.find( "a" )
```

Comment:
We've documented `_buttonHtml` as an extension point: http://api.jqueryui.com/spinner/#method-_buttonHtml

The example returns `<button>`, so this change isn't backwards compatible. I'll leave it at that for now, we can discuss this elsewhere or ignore it for now, land this PR and revisit spinner afterwards.